tests/test_sshfsmount.cpp

```diff
@@ -188,167 +227,173 @@ struct SshfsMount : public mp::test::SftpServerTest
+CommandVector old_fuse_cmds = {{"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -V", "FUSE library version: 2.9.0\n"},
+                               {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -o slave -o transform_symlinks -o "
+                                "allow_other -o nonempty :\"source\" \"target\"",
+                                "don't care\n"}};
+
+// Commands to check that a version of FUSE at least 3.0.0 gives a correct answer.
+CommandVector new_fuse_cmds = {{"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -V", "FUSE library version: 3.0.0\n"},
+                               {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -o slave -o transform_symlinks -o "
+                                "allow_other :\"source\" \"target\"",
+                                "don't care\n"}};
+
+// Commands to check that the server correctly creates the mount target.
+CommandVector exec_cmds = {
+    {"sudo /bin/bash -c 'P=\"/home/ubuntu/target\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'",
+     "/home/ubuntu/\n"},
+    {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && mkdir -p \"target\"'", "\n"},
+    {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && chown -R 1000:1000 \"target\"'", "\n"}};
+
+// Commands to check that the server works if an absolute path is given.
+CommandVector absolute_cmds = {
+    {"sudo /bin/bash -c 'P=\"/home/ubuntu/target\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'",
+     "/home/ubuntu/\n"}};
+
+// Commands to check that it works for a nonexisting path.
+CommandVector nonexisting_path_cmds = {
+    {"sudo /bin/bash -c 'P=\"/nonexisting/path\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'", "/\n"},
+    {"sudo /bin/bash -c 'cd \"/\" && mkdir -p \"nonexisting/path\"'", "\n"},
+    {"sudo /bin/bash -c 'cd \"/\" && chown -R 1000:1000 \"nonexisting\"'", "\n"}};
+
+// Check the execution of the CommandVector's above.
+INSTANTIATE_TEST_SUITE_P(SshfsMountSuccess, SshfsMountExecute,
+                         testing::Values(std::make_pair("target", old_fuse_cmds), std::make_pair("target", exec_cmds),
+                                         std::make_pair("target", new_fuse_cmds), std::make_pair("target", exec_cmds),
+                                         std::make_pair("/home/ubuntu/target", absolute_cmds),
+                                         std::make_pair("/nonexisting/path", nonexisting_path_cmds)));
+
+// Commands to test that when a mount path already exists, no mkdir nor chown is ran.
+CommandVector execute_no_mkdir_cmds = {
+    {"sudo /bin/bash -c 'P=\"/home/ubuntu/target\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'",
+     "/home/ubuntu/target/\n"}};
+
+INSTANTIATE_TEST_SUITE_P(SshfsMountSuccessAndAvoidCommands, SshfsMountExecuteAndFail,
+                         testing::Values(std::make_tuple("target", execute_no_mkdir_cmds, "mkdir"),
+                                         std::make_tuple("target", execute_no_mkdir_cmds, "chown")));
```

Comment:
So this is the case that you said needs fixing? I don't think we should be adding a test that tests for wrong behaviour :)

It'd be good to fix in this PR, IMO.

Reply:
Yes, `make_exec_that_fails` is a special case, I refactored it.

On fixing the `mkdir`, `chown` issue on this PR, yes, it makes sense to fix it here. In fact, the problem was that when no directory must be created, the 'missing' part of the path is `.`. This makes sense because it's the output of `QDir::relativeFilePath()`. I added so far a check in `src/sshfs_mount/sshfs_mount.cpp`. Now I moved it to the correct place: if missing is `.`, then no folder must be created nor changed its owner.